File: mobile/src/main/java/com/chernowii/camcontrol/camera/goproAPI/GPConstants.java
package com.chernowii.camcontrol.camera.goproAPI;


import java.net.URI;

/**
 * Created by konrad on 1/2/17.
 */

public class GPConstants {
    public static String settingsUri(String param, String val) {
        return "http://10.5.5.9/gp/gpControl/setting/" + param + "/" + val;
    }
    public static String commandUri(String param) {
        return "http://10.5.5.9/gp/gpControl/command/" + param;
    }
    public static String executeUri(String param) {
        return "http://10.5.5.9/gp/gpControl/execute" + param;
    }
    public static class Commands {
    public static class Stream{
        public static URI Restart = URI.create(executeUri("?p1=gpStream&a1=proto_v2&c1=restart"));
    }
    public static class Shutter {
        public static URI shutter = URI.create(commandUri("shutter?p=1"));
        public static URI stop = URI.create(commandUri("shutter?p=0"));
        public static URI hilight = URI.create(commandUri("storage/tag_moment"));
    }

    public static class Modes {
        public static URI videoMode = URI.create(commandUri("mode?p=0"));
        public static URI photoMode = URI.create(commandUri("mode?p=1"));
        public static URI multishotMode = URI.create(commandUri("mode?p=2"));
        public static class videoSubModes {
            public static URI Video = URI.create(commandUri("sub_mode?mode=0&sub_mode=0"));
            public static URI TimeLapseVideo = URI.create(commandUri("sub_mode?mode=0&sub_mode=1"));
            public static URI VideoPlusPhoto = URI.create(commandUri("sub_mode?mode=0&sub_mode=2"));
            public static URI Looping = URI.create(commandUri("sub_mode?mode=0&sub_mode=3"));
        }

        public static class photoSubModes {
            public static URI Single = URI.create(commandUri("sub_mode?mode=1&sub_mode=0"));
            public static URI Continuous = URI.create(commandUri("sub_mode?mode=1&sub_mode=1"));
            public static URI Night = URI.create(commandUri("sub_mode?mode=1&sub_mode=2"));
        }

        public static class multiShotSubModes {
            public static URI Burst = URI.create(commandUri("sub_mode?mode=2&sub_mode=0"));
            public static URI Timelapse = URI.create(commandUri("sub_mode?mode=2&sub_mode=1"));
            public static URI NightLapse = URI.create(commandUri("sub_mode?mode=2&sub_mode=2"));
        }
    }


}
    public static class Setup {
        public static class Orientation {
            public static URI Up = URI.create(settingsUri("52", "1"));
            public static URI Down = URI.create(settingsUri("52", "2"));
            public static URI Auto = URI.create(settingsUri("52", "0"));
        }

        public static class Locate {

            public static URI On = URI.create(commandUri("system/locate?p=1"));
            public static URI Off = URI.create(commandUri("system/locate?p=0"));

        }

        public static class Delete {
            public static URI DeleteFile = URI.create(commandUri("storage/delete?p=file"));
            public static URI DeleteLast = URI.create(commandUri("storage/delete/last"));
            public static URI Format = URI.create(commandUri("storage/delete/all"));
        }

        public static class QuickCapture {
            public static URI On = URI.create(settingsUri("54", "1"));
            public static URI Off = URI.create(settingsUri("54", "0"));

        }

        public static class LEDBlink {
            public static URI Two = URI.create(settingsUri("55", "1"));
            public static URI Four = URI.create(settingsUri("55", "2"));
            public static URI Off = URI.create(settingsUri("55", "0"));

        }

        public static class Beeps {
            public static URI Off = URI.create(settingsUri("56", "2"));
            public static URI SeventyPercent = URI.create(settingsUri("56", "1"));
            public static URI Full = URI.create(settingsUri("56", "0"));

        }

        public static class VideoFormat {
            public static URI NTSC = URI.create(settingsUri("57", "0"));
            public static URI PAL = URI.create(settingsUri("57", "1"));

        }

        public static class LCDDisplay {
            public static URI On = URI.create(settingsUri("72", "1"));
            public static URI Off = URI.create(settingsUri("72", "0"));

        }

        public static class OnScreenDisplay {
            public static URI On = URI.create(settingsUri("58", "1"));
            public static URI Off = URI.create(settingsUri("58", "0"));

        }

        public static class LCDBrightness {
            public static URI High = URI.create(settingsUri("49", "0"));
            public static URI Medium = URI.create(settingsUri("49", "1"));
            public static URI Low = URI.create(settingsUri("49", "2"));

        }

        public static class LCDLock {
            public static URI On = URI.create(settingsUri("50", "1"));
            public static URI Off = URI.create(settingsUri("50", "0"));

        }

        public static class LCDTimeout {

            public static URI Never = URI.create(settingsUri("51", "0"));
            public static URI LCD1min = URI.create(settingsUri("51", "1"));
            public static URI LCD2min = URI.create(settingsUri("51", "2"));
            public static URI LCD3min = URI.create(settingsUri("51", "3"));

        }

        public static class AutoOff {
            public static URI Never = URI.create(settingsUri("59", "0"));
            public static URI AO1m = URI.create(settingsUri("59", "1"));
            public static URI AO2m = URI.create(settingsUri("59", "2"));
            public static URI AO3m = URI.create(settingsUri("59", "3"));
            public static URI AO5m = URI.create(settingsUri("59", "4"));

        }
    }
    public static class Video{
        public static class Resolutions {
            public static URI Res4K= URI.create(settingsUri("2","1"));
            public static URI Res4K_SuperView= URI.create(settingsUri("2","2"));
            public static URI Res2point7K= URI.create(settingsUri("2","4"));
            public static URI Res2point7KSuperView= URI.create(settingsUri("2","5"));
            public static URI Res2point7K_4by3AR= URI.create(settingsUri("2","6"));
            public static URI Res1440p= URI.create(settingsUri("2","7"));
            public static URI Res1080pSuperView= URI.create(settingsUri("2","8"));
            public static URI Res1080p= URI.create(settingsUri("2","9"));
            public static URI Res960p= URI.create(settingsUri("2","10"));
            public static URI Res720pSuperView= URI.create(settingsUri("2","11"));
            public static URI Res720p= URI.create(settingsUri("2","12"));
            public static URI ResWVGA= URI.create(settingsUri("2","13"));
        }
        public static class FrameRate {
            public static URI FR240fps= URI.create(settingsUri("3","0"));
            public static URI FR120fps= URI.create(settingsUri("3","1"));
            public static URI FR100fps= URI.create(settingsUri("3","2"));
            public static URI FR60fps= URI.create(settingsUri("3","5"));
            public static URI FR50fps= URI.create(settingsUri("3","6"));
            public static URI FR48fps= URI.create(settingsUri("3","7"));
            public static URI FR30fps= URI.create(settingsUri("3","8"));
            public static URI FR25fps= URI.create(settingsUri("3","9"));
            public static URI FR24fps= URI.create(settingsUri("3","10"));
            public static URI FR15fps= URI.create(settingsUri("3","11"));
            public static URI FR12point5fps= URI.create(settingsUri("3","12"));
        }
        public static class FOV {

            public static URI Wide= URI.create(settingsUri("4","0"));
            public static URI Medium= URI.create(settingsUri("4","1"));
            public static URI Narrow= URI.create(settingsUri("4","2"));
            public static URI Linear= URI.create(settingsUri("4","4"));
        }
        public static class LowLight{
            public static URI On= URI.create(settingsUri("8","1"));
            public static URI Off= URI.create(settingsUri("8","0"));
        }
        public static class LoopingDuration{

            public static URI Max= URI.create(settingsUri("6","0"));
            public static URI Loop5Min= URI.create(settingsUri("6","1"));
            public static URI Loop20Min= URI.create(settingsUri("6","2"));
            public static URI Loop60Min= URI.create(settingsUri("6","3"));
            public static URI Loop120Min= URI.create(settingsUri("6","4"));
        }
        public static class VideoPlusPhotoInterval{

            public static URI Interval5Min= URI.create(settingsUri("7","1"));
            public static URI Interval10Min= URI.create(settingsUri("7","2"));
            public static URI Interval30Min= URI.create(settingsUri("7","3"));
            public static URI Interval60Min= URI.create(settingsUri("7","4"));
        }
        public static class SpotMeter {

            public static URI Off = URI.create(settingsUri("9", "0"));
            public static URI On = URI.create(settingsUri("9", "1"));
        }
        public static class ProTune{
            public static URI On = URI.create(settingsUri("10","1"));
            public static URI Off = URI.create(settingsUri("10","0"));
        }
        public static class WhiteBalance{

            public static URI WBAuto = URI.create(settingsUri("11","0"));
            public static URI WB3000k = URI.create(settingsUri("11","1"));
            public static URI WB4000k = URI.create(settingsUri("11","5"));
            public static URI WB4800k = URI.create(settingsUri("11","6"));
            public static URI WB5500k = URI.create(settingsUri("11","2"));
            public static URI WB6000k = URI.create(settingsUri("11","7"));
            public static URI WB6500k = URI.create(settingsUri("11","3"));
            public static URI WBNative = URI.create(settingsUri("11","4"));
        }
        public static class Color{

            public static URI GOPRO = URI.create(settingsUri("12","0"));
            public static URI Flat = URI.create(settingsUri("12","1"));
        }
        public static class ISOLimit{
            public static URI ISO6400 = URI.create(settingsUri("13","0"));
            public static URI ISO1600 = URI.create(settingsUri("13","1"));
            public static URI ISO400 = URI.create(settingsUri("13","2"));
            public static URI ISO3200 = URI.create(settingsUri("13","3"));
            public static URI ISO800 = URI.create(settingsUri("13","4"));
            public static URI ISO200 = URI.create(settingsUri("13","7"));
            public static URI ISO100 = URI.create(settingsUri("13","8"));
        }
        public static class ISOMode{
            public static URI Max = URI.create(settingsUri("74","0"));
            public static URI Lock = URI.create(settingsUri("74","1"));
        }
        public static class Sharpness{
            public static URI High = URI.create(settingsUri("14","0"));
            public static URI Med = URI.create(settingsUri("14","1"));
            public static URI Low = URI.create(settingsUri("14","2"));
        }
        public static class ManualVideoExposure{
            public static URI MVE_AutoMode = URI.create(settingsUri("73","0"));
            public static class FR24FPS{
                public static URI MVE_1_24 = URI.create(settingsUri("73","3"));
                public static URI MVE_1_48 = URI.create(settingsUri("73","6"));
                public static URI MVE_1_96 = URI.create(settingsUri("73","11"));
            }
            public static class FR30FPS{
                public static URI MVE_1_30 = URI.create(settingsUri("73","5"));
                public static URI MVE_1_60 = URI.create(settingsUri("73","8"));
                public static URI MVE_1_120 = URI.create(settingsUri("73","13"));
            }
            public static class FR48FPS{
                public static URI MVE_1_48 = URI.create(settingsUri("73","6"));
                public static URI MVE_1_96 = URI.create(settingsUri("73","11"));
                public static URI MVE_1_192 = URI.create(settingsUri("73","16"));
            }
            public static class FR60FPS{
                public static URI MVE_1_60 = URI.create(settingsUri("73","8"));
                public static URI MVE_1_120 = URI.create(settingsUri("73","13"));
                public static URI MVE_1_240 = URI.create(settingsUri("73","18"));
            }
            public static class FR90FPS{
                public static URI MVE_1_90 = URI.create(settingsUri("73","10"));
                public static URI MVE_1_180 = URI.create(settingsUri("73","15"));
                public static URI MVE_1_360 = URI.create(settingsUri("73","20"));
            }
            public static class FR120FPS{
                public static URI MVE_1_120 = URI.create(settingsUri("73","13"));
                public static URI MVE_1_240 = URI.create(settingsUri("73","18"));
                public static URI MVE_1_480 = URI.create(settingsUri("73","22"));
            }
            public static class FR240FPS{
                public static URI MVE_1_120 = URI.create(settingsUri("73","18"));
                public static URI MVE_1_240 = URI.create(settingsUri("73","22"));
                public static URI MVE_1_480 = URI.create(settingsUri("73","23"));

            }
        }
        public static class EVComp{
            static String param="15";
            public static URI EVMinus2_0 = URI.create(settingsUri(param,"8"));
            public static URI EVMinus1_5 = URI.create(settingsUri(param,"7"));
            public static URI EVMinus1_0 = URI.create(settingsUri(param,"6"));
            public static URI EVMinus0_5 = URI.create(settingsUri(param,"5"));
            public static URI EV0_0 = URI.create(settingsUri(param,"4"));
            public static URI EV0_5 = URI.create(settingsUri(param,"3"));
            public static URI EV1_0 = URI.create(settingsUri(param,"2"));
            public static URI EV1_5 = URI.create(settingsUri(param,"1"));
            public static URI EV2_0 = URI.create(settingsUri(param,"0"));

        }
    }

    public static class Photo{

        public static class ContinuousRate{
            static String param="18";
            public static URI C3PPS = URI.create(settingsUri(param,"0"));
            public static URI C5PPS = URI.create(settingsUri(param,"1"));
            public static URI C10PPS = URI.create(settingsUri(param,"2"));

        }

        public static class Resolution{
            static String param="17";
            public static URI R12MPW = URI.create(settingsUri(param,"0"));
            public static URI R12MPL= URI.create(settingsUri(param,"10"));
            public static URI R12MPM= URI.create(settingsUri(param,"8"));
            public static URI R12MPN= URI.create(settingsUri(param,"9"));
            public static URI R10MPW= URI.create(settingsUri(param,"4"));
            public static URI R10MPL= URI.create(settingsUri(param,"11"));
            public static URI R8MPW = URI.create(settingsUri(param,"0"));
            public static URI R7MPW = URI.create(settingsUri(param,"1"));
            public static URI R7MPM = URI.create(settingsUri(param,"2"));
            public static URI R5MPM = URI.create(settingsUri(param,"3"));

        }
        public static class Shutter{
            static String param="19";
            public static URI ShutterAuto = URI.create(settingsUri(param,"0"));
            public static URI Shutter2S = URI.create(settingsUri(param,"1"));
            public static URI Shutter5S = URI.create(settingsUri(param,"2"));
            public static URI Shutter10S = URI.create(settingsUri(param,"3"));
            public static URI Shutter15S = URI.create(settingsUri(param,"4"));
            public static URI Shutter20S = URI.create(settingsUri(param,"5"));
            public static URI Shutter30S = URI.create(settingsUri(param,"6"));

        }
        public static class SpotMeter{
            static String param="20";
            public static URI ON = URI.create(settingsUri(param,"1"));
            public static URI OFF = URI.create(settingsUri(param,"0"));

        }
        public static class WDR{
            static String param="77";
            public static URI ON = URI.create(settingsUri(param,"1"));
            public static URI OFF = URI.create(settingsUri(param,"0"));

        }
        public static class RAW{
            static String param="82";
            public static URI ON = URI.create(settingsUri(param,"1"));
            public static URI OFF = URI.create(settingsUri(param,"0"));

        }
        public static class Protune{
            static String param="21";
            public static URI ON = URI.create(settingsUri(param,"1"));
            public static URI OFF = URI.create(settingsUri(param,"0"));

        }
        public static class WhiteBalance{
            static String param="22";
            public static URI WBAuto = URI.create(settingsUri(param,"0"));
            public static URI WB3000K = URI.create(settingsUri(param,"1"));
            public static URI WB4000K = URI.create(settingsUri(param,"5"));
            public static URI WB4800K = URI.create(settingsUri(param,"6"));
            public static URI WB5500K = URI.create(settingsUri(param,"2"));
            public static URI WB6000K = URI.create(settingsUri(param,"7"));
            public static URI WB6500K = URI.create(settingsUri(param,"3"));
            public static URI WBNative = URI.create(settingsUri(param,"4"));

        }
        public static class Color{
            static String param="23";
            public static URI GoProColor = URI.create(settingsUri(param,"0"));
            public static URI Flat = URI.create(settingsUri(param,"1"));

        }
        public static class Sharpness{
            static String param="25";
            public static URI High = URI.create(settingsUri(param,"0"));
            public static URI Medium = URI.create(settingsUri(param,"1"));
            public static URI Low = URI.create(settingsUri(param,"2"));

        }
        public static class EVComp{
            static String param="26";
            public static URI EVMinus2_0 = URI.create(settingsUri(param,"8"));
            public static URI EVMinus1_5 = URI.create(settingsUri(param,"7"));
            public static URI EVMinus1_0 = URI.create(settingsUri(param,"6"));
            public static URI EVMinus0_5 = URI.create(settingsUri(param,"5"));
            public static URI EV0_0 = URI.create(settingsUri(param,"4"));
            public static URI EV0_5 = URI.create(settingsUri(param,"3"));
            public static URI EV1_0 = URI.create(settingsUri(param,"2"));
            public static URI EV1_5 = URI.create(settingsUri(param,"1"));
            public static URI EV2_0 = URI.create(settingsUri(param,"0"));

        }
        public static class ISOMin{
            static String param="75";
            public static URI ISO800 = URI.create(settingsUri(param,"0"));
            public static URI ISO400 = URI.create(settingsUri(param,"1"));
            public static URI ISO200 = URI.create(settingsUri(param,"2"));
            public static URI ISO100 = URI.create(settingsUri(param,"3"));

        }
        public static class ISOMax{
            static String param="24";
            public static URI ISO800 = URI.create(settingsUri(param,"0"));
            public static URI ISO400 = URI.create(settingsUri(param,"1"));
            public static URI ISO200 = URI.create(settingsUri(param,"2"));
            public static URI ISO100 = URI.create(settingsUri(param,"3"));
        }
    }
}

File: mobile/src/main/java/com/chernowii/camcontrol/camera/goproAPI/model/GPConstants.kt
package com.chernowii.camcontrol.camera.goproAPI.model


import com.chernowii.camcontrol.camera.goproAPI.ApiBase
import com.chernowii.camcontrol.camera.goproAPI.ApiClient

import java.net.URI

import retrofit2.Call

/**
 * Created by konrad on 1/2/17.
 */

object GPConstants {
    internal var GoProApi = ApiBase.mainClient.create(ApiClient::class.java!!)

    class Commands {
        object Stream {
            var Restart = GoProApi.execute("?p1=gpStream&a1=proto_v2&c1=restart")
        }

        object Shutter {
            var shutter = GoProApi.command("shutter", "1")
            var stop = GoProApi.command("shutter", "0")
            var hilight = GoProApi.command("storage/tag_moment", "")
        }

        object Modes {
            var videoMode = GoProApi.command("mode", "0")
            var photoMode = GoProApi.command("mode", "1")
            var multishotMode = GoProApi.command("mode", "2")

            object videoSubModes {
                var Video = GoProApi.command("sub_mode", "mode=0&sub_mode=0")
                var TimeLapseVideo = GoProApi.command("sub_mode", "mode=0&sub_mode=1")
                var VideoPlusPhoto = GoProApi.command("sub_mode", "mode=0&sub_mode=2")
                var Looping = GoProApi.command("sub_mode", "mode=0&sub_mode=3")
            }

            object photoSubModes {
                var Single = GoProApi.command("sub_mode", "mode=1&sub_mode=0")
                var Continuous = GoProApi.command("sub_mode", "mode=1&sub_mode=1")
                var Night = GoProApi.command("sub_mode", "mode=1&sub_mode=2")
            }

            object multiShotSubModes {
                var Burst = GoProApi.command("sub_mode", "mode=2&sub_mode=0")
                var Timelapse = GoProApi.command("sub_mode", "mode=2&sub_mode=1")
                var NightLapse = GoProApi.command("sub_mode", "mode=2&sub_mode=2")
            }
        }


    }

    class Setup {
        object Orientation {
            var Up = GoProApi.config("52", "1")
            var Down = GoProApi.config("52", "2")
            var Auto = GoProApi.config("52", "0")
        }

        object Locate {

            var On = GoProApi.command("system/locate", "1")
            var Off = GoProApi.command("system/locate", "0")

        }

        object Delete {
            var DeleteFile = GoProApi.command("storage/delete", "file")
            var DeleteLast = GoProApi.command("storage/delete/last", "")
            var Format = GoProApi.command("storage/delete/all", "")
        }

        object QuickCapture {
            var On = GoProApi.config("54", "1")
            var Off = GoProApi.config("54", "0")

        }

        object LEDBlink {
            var Two = GoProApi.config("55", "1")
            var Four = GoProApi.config("55", "2")
            var Off = GoProApi.config("55", "0")

        }

        object Beeps {
            var Off = GoProApi.config("56", "2")
            var SeventyPercent = GoProApi.config("56", "1")
            var Full = GoProApi.config("56", "0")

        }

        object VideoFormat {
            var NTSC = GoProApi.config("57", "0")
            var PAL = GoProApi.config("57", "1")

        }

        object LCDDisplay {
            var On = GoProApi.config("72", "1")
            var Off = GoProApi.config("72", "0")

        }

        object OnScreenDisplay {
            var On = GoProApi.config("58", "1")
            var Off = GoProApi.config("58", "0")

        }

        object LCDBrightness {
            var High = GoProApi.config("49", "0")
            var Medium = GoProApi.config("49", "1")
            var Low = GoProApi.config("49", "2")

        }

        object LCDLock {
            var On = GoProApi.config("50", "1")
            var Off = GoProApi.config("50", "0")

        }

        object LCDTimeout {

            var Never = GoProApi.config("51", "0")
            var LCD1min = GoProApi.config("51", "1")
            var LCD2min = GoProApi.config("51", "2")
            var LCD3min = GoProApi.config("51", "3")

        }

        object AutoOff {
            var Never = GoProApi.config("59", "0")
            var AO1m = GoProApi.config("59", "1")
            var AO2m = GoProApi.config("59", "2")
            var AO3m = GoProApi.config("59", "3")
            var AO5m = GoProApi.config("59", "4")

        }
    }

    class Video {
        object Resolutions {
            var Res4K = GoProApi.config("2", "1")
            var Res4K_SuperView = GoProApi.config("2", "2")
            var Res2point7K = GoProApi.config("2", "4")
            var Res2point7KSuperView = GoProApi.config("2", "5")
            var Res2point7K_4by3AR = GoProApi.config("2", "6")
            var Res1440p = GoProApi.config("2", "7")
            var Res1080pSuperView = GoProApi.config("2", "8")
            var Res1080p = GoProApi.config("2", "9")
            var Res960p = GoProApi.config("2", "10")
            var Res720pSuperView = GoProApi.config("2", "11")
            var Res720p = GoProApi.config("2", "12")
            var ResWVGA = GoProApi.config("2", "13")
        }

        object FrameRate {
            var FR240fps = GoProApi.config("3", "0")
            var FR120fps = GoProApi.config("3", "1")
            var FR100fps = GoProApi.config("3", "2")
            var FR60fps = GoProApi.config("3", "5")
            var FR50fps = GoProApi.config("3", "6")
            var FR48fps = GoProApi.config("3", "7")
            var FR30fps = GoProApi.config("3", "8")
            var FR25fps = GoProApi.config("3", "9")
            var FR24fps = GoProApi.config("3", "10")
            var FR15fps = GoProApi.config("3", "11")
            var FR12point5fps = GoProApi.config("3", "12")
        }

        object FOV {

            var Wide = GoProApi.config("4", "0")
            var Medium = GoProApi.config("4", "1")
            var Narrow = GoProApi.config("4", "2")
            var Linear = GoProApi.config("4", "4")
        }

        object LowLight {
            var On = GoProApi.config("8", "1")
            var Off = GoProApi.config("8", "0")
        }

        object LoopingDuration {

            var Max = GoProApi.config("6", "0")
            var Loop5Min = GoProApi.config("6", "1")
            var Loop20Min = GoProApi.config("6", "2")
            var Loop60Min = GoProApi.config("6", "3")
            var Loop120Min = GoProApi.config("6", "4")
        }

        object VideoPlusPhotoInterval {

            var Interval5Min = GoProApi.config("7", "1")
            var Interval10Min = GoProApi.config("7", "2")
            var Interval30Min = GoProApi.config("7", "3")
            var Interval60Min = GoProApi.config("7", "4")
        }

        object SpotMeter {

            var Off = GoProApi.config("9", "0")
            var On = GoProApi.config("9", "1")
        }

        object ProTune {
            var On = GoProApi.config("10", "1")
            var Off = GoProApi.config("10", "0")
        }

        object WhiteBalance {

            var WBAuto = GoProApi.config("11", "0")
            var WB3000k = GoProApi.config("11", "1")
            var WB4000k = GoProApi.config("11", "5")
            var WB4800k = GoProApi.config("11", "6")
            var WB5500k = GoProApi.config("11", "2")
            var WB6000k = GoProApi.config("11", "7")
            var WB6500k = GoProApi.config("11", "3")
            var WBNative = GoProApi.config("11", "4")
        }

        object Color {

            var GOPRO = GoProApi.config("12", "0")
            var Flat = GoProApi.config("12", "1")
        }

        object ISOLimit {
            var ISO6400 = GoProApi.config("13", "0")
            var ISO1600 = GoProApi.config("13", "1")
            var ISO400 = GoProApi.config("13", "2")
            var ISO3200 = GoProApi.config("13", "3")
            var ISO800 = GoProApi.config("13", "4")
            var ISO200 = GoProApi.config("13", "7")
            var ISO100 = GoProApi.config("13", "8")
        }

        object ISOMode {
            var Max = GoProApi.config("74", "0")
            var Lock = GoProApi.config("74", "1")
        }

        object Sharpness {
            var High = GoProApi.config("14", "0")
            var Med = GoProApi.config("14", "1")
            var Low = GoProApi.config("14", "2")
        }

        object ManualVideoExposure {
            var MVE_AutoMode = GoProApi.config("73", "0")

            object FR24FPS {
                var MVE_1_24 = GoProApi.config("73", "3")
                var MVE_1_48 = GoProApi.config("73", "6")
                var MVE_1_96 = GoProApi.config("73", "11")
            }

            object FR30FPS {
                var MVE_1_30 = GoProApi.config("73", "5")
                var MVE_1_60 = GoProApi.config("73", "8")
                var MVE_1_120 = GoProApi.config("73", "13")
            }

            object FR48FPS {
                var MVE_1_48 = GoProApi.config("73", "6")
                var MVE_1_96 = GoProApi.config("73", "11")
                var MVE_1_192 = GoProApi.config("73", "16")
            }

            object FR60FPS {
                var MVE_1_60 = GoProApi.config("73", "8")
                var MVE_1_120 = GoProApi.config("73", "13")
                var MVE_1_240 = GoProApi.config("73", "18")
            }

            object FR90FPS {
                var MVE_1_90 = GoProApi.config("73", "10")
                var MVE_1_180 = GoProApi.config("73", "15")
                var MVE_1_360 = GoProApi.config("73", "20")
            }

            object FR120FPS {
                var MVE_1_120 = GoProApi.config("73", "13")
                var MVE_1_240 = GoProApi.config("73", "18")
                var MVE_1_480 = GoProApi.config("73", "22")
            }

            object FR240FPS {
                var MVE_1_120 = GoProApi.config("73", "18")
                var MVE_1_240 = GoProApi.config("73", "22")
                var MVE_1_480 = GoProApi.config("73", "23")

            }
        }

        object EVComp {
            internal var param = "15"
            var EVMinus2_0 = GoProApi.config(param, "8")
            var EVMinus1_5 = GoProApi.config(param, "7")
            var EVMinus1_0 = GoProApi.config(param, "6")
            var EVMinus0_5 = GoProApi.config(param, "5")
            var EV0_0 = GoProApi.config(param, "4")
            var EV0_5 = GoProApi.config(param, "3")
            var EV1_0 = GoProApi.config(param, "2")
            var EV1_5 = GoProApi.config(param, "1")
            var EV2_0 = GoProApi.config(param, "0")

        }
    }

    class Photo {

        object ContinuousRate {
            internal var param = "18"
            var C3PPS = GoProApi.config(param, "0")
            var C5PPS = GoProApi.config(param, "1")
            var C10PPS = GoProApi.config(param, "2")

        }

        object Resolution {
            internal var param = "17"
            var R12MPW = GoProApi.config(param, "0")
            var R12MPL = GoProApi.config(param, "10")
            var R12MPM = GoProApi.config(param, "8")
            var R12MPN = GoProApi.config(param, "9")
            var R10MPW = GoProApi.config(param, "4")
            var R10MPL = GoProApi.config(param, "11")
            var R8MPW = GoProApi.config(param, "0")
            var R7MPW = GoProApi.config(param, "1")
            var R7MPM = GoProApi.config(param, "2")
            var R5MPM = GoProApi.config(param, "3")

        }

        object Shutter {
            internal var param = "19"
            var ShutterAuto = GoProApi.config(param, "0")
            var Shutter2S = GoProApi.config(param, "1")
            var Shutter5S = GoProApi.config(param, "2")
            var Shutter10S = GoProApi.config(param, "3")
            var Shutter15S = GoProApi.config(param, "4")
            var Shutter20S = GoProApi.config(param, "5")
            var Shutter30S = GoProApi.config(param, "6")

        }

        object SpotMeter {
            internal var param = "20"
            var ON = GoProApi.config(param, "1")
            var OFF = GoProApi.config(param, "0")

        }

        object WDR {
            internal var param = "77"
            var ON = GoProApi.config(param, "1")
            var OFF = GoProApi.config(param, "0")

        }

        object RAW {
            internal var param = "82"
            var ON = GoProApi.config(param, "1")
            var OFF = GoProApi.config(param, "0")

        }

        object Protune {
            internal var param = "21"
            var ON = GoProApi.config(param, "1")
            var OFF = GoProApi.config(param, "0")

        }

        object WhiteBalance {
            internal var param = "22"
            var WBAuto = GoProApi.config(param, "0")
            var WB3000K = GoProApi.config(param, "1")
            var WB4000K = GoProApi.config(param, "5")
            var WB4800K = GoProApi.config(param, "6")
            var WB5500K = GoProApi.config(param, "2")
            var WB6000K = GoProApi.config(param, "7")
            var WB6500K = GoProApi.config(param, "3")
            var WBNative = GoProApi.config(param, "4")

        }

        object Color {
            internal var param = "23"
            var GoProColor = GoProApi.config(param, "0")
            var Flat = GoProApi.config(param, "1")

        }

        object Sharpness {
            internal var param = "25"
            var High = GoProApi.config(param, "0")
            var Medium = GoProApi.config(param, "1")
            var Low = GoProApi.config(param, "2")

        }

        object EVComp {
            internal var param = "26"
            var EVMinus2_0 = GoProApi.config(param, "8")
            var EVMinus1_5 = GoProApi.config(param, "7")
            var EVMinus1_0 = GoProApi.config(param, "6")
            var EVMinus0_5 = GoProApi.config(param, "5")
            var EV0_0 = GoProApi.config(param, "4")
            var EV0_5 = GoProApi.config(param, "3")
            var EV1_0 = GoProApi.config(param, "2")
            var EV1_5 = GoProApi.config(param, "1")
            var EV2_0 = GoProApi.config(param, "0")

        }

        object ISOMin {
            internal var param = "75"
            var ISO800 = GoProApi.config(param, "0")
            var ISO400 = GoProApi.config(param, "1")
            var ISO200 = GoProApi.config(param, "2")
            var ISO100 = GoProApi.config(param, "3")

        }

        object ISOMax {
            internal var param = "24"
            var ISO800 = GoProApi.config(param, "0")
            var ISO400 = GoProApi.config(param, "1")
            var ISO200 = GoProApi.config(param, "2")
            var ISO100 = GoProApi.config(param, "3")
        }
    }
}
